service/internal/configunmarshaler/dry_run_unmarshaller.go

```diff
@@ -0,0 +1,233 @@
+package configunmarshaler
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type ExtensionValidateError struct {
```

Comment:
If those four types have the same definition, perhaps you can have only one? `ComponentValidationError` would probably be a good candidate here.